pkgs/ffigen/lib/src/header_parser/sub_parsers/api_availability.dart

```diff
@@ -80,52 +96,42 @@ class ApiAvailability {
     return api;
   }
 
-  bool isAvailable(ExternalVersions extVers) {
-    final macosVer = extVers.macos?.min;
-    final iosVer = extVers.ios?.min;
+  Availability getAvailability(ExternalVersions extVers) {
+    final macosVer = _normalizeVersions(extVers.macos);
+    final iosVer = _normalizeVersions(extVers.ios);
 
     // If no versions are specified, everything is available.
     if (iosVer == null && macosVer == null) {
-      return true;
+      return Availability.all;
     }
 
     if (alwaysDeprecated || alwaysUnavailable) {
-      return false;
+      return Availability.none;
     }
 
-    for (final (plat, minVer) in [(ios, iosVer), (macos, macosVer)]) {
-      // If the user hasn't specified a minimum version for this platform, defer
-      // to the other platforms.
-      if (minVer == null) {
+    Availability? availability;
+    for (final (plat, ver) in [(ios, iosVer), (macos, macosVer)]) {
```

Comment:
Not much more characters to write `platform` and `version` and it follows the dart style.

Reply:
Done